index.bs

```diff
@@ -4109,165 +4301,1226 @@ To <dfn>get the response data</dfn> given |response|:
+     <dd>|url|'s [=url/scheme=] if |has protocol| is true, or null otherwise.
+     <dt>hostname
+     <dd>|url|'s [=url/host=] if |has hostname| is true, or null otherwise.
+     <dt>port
+     <dd>[=Serialize an integer=] with |url|'s [=url/port=], if it's not null,
+         or null otherwise.
+     <dt>pathname
+     <dd>The result of running the [=URL path serializer=] with |url|'s
+         [=url/path=], if it's not [=list/empty=], or null otherwise.
+     <dt>search
+     <dd>|url|'s [=url/query=].
+   </dl>
+
+1. Return [=success=] with data |parsed|.
+
+</div>
+
+<div algorithm>
+To <dfn>match URL pattern</dfn> given |url pattern| and |url string|:
+
+1. Let |url| be the result of [=URL parser|parsing=] |url string|.
+
+1. If |url pattern|'s protocol is not null and is not equal to |url|'s
+   [=url/scheme=], return false.
+
+1. If |url pattern|'s hostname is not null and is not equal to |url|'s
+   [=url/host=], return false.
+
+1. If |url pattern|'s port is not null:
+
+   1. Let |port number| be |url|'s [=url/port=] if it's not null, or the
+      [=default port=] for |url|'s [=url/scheme=] otherwise.
+
+   1. If |port number| is null, return false.
+
+   1. If |url pattern|'s port is not equal to [=serialize an integer=] with
+      |port number|, return false.
+
+1. If |url pattern|'s pathname is not null and is not equal to the result of
+   running the [=URL path serializer=] with |url|'s [=url/path=], return false.
+
+1. If |url pattern|'s search is not null and is not equal to |url|'s
+   [=url/query=], return false.
+
+1. Return true.
+
+</div>
+
+### Commands ### {#module-network-commands}
+
+#### The network.addIntercept Command ####  {#command-network-addIntercept}
+
+The <dfn export for=commands>network.addIntercept</dfn> command adds a
+[=network intercept=].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      network.AddIntercept = {
+        method: "network.addIntercept",
+        params: network.AddInterceptParameters
+      }
+
+      network.AddInterceptParameters = {
+        phases: [+network.InterceptPhase],
+        ?urlPatterns: [*network.UrlPattern],
+      }
+
+      network.InterceptPhase = "beforeRequestSent" / "responseStarted" /
+                               "authRequired"
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      network.AddInterceptResult = {
+        intercept: network.Intercept
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for network.addRequestIntercept">
+The [=remote end steps=] given |session| and |command parameters| are:
+
+1. Let |intercept| be the string representation of a [[!RFC4122|UUID]].
+
+1. Let |url patterns| be the <code>urlPatterns</code> field of |command
+   parameters| if present, or null otherwise.
+
+1. Let |intercept map| be |session|'s [=intercept map=].
+
+1. Let |parsed patterns| be an empty list.
+
+1. For each |url pattern| in |url patterns|:
```

Comment:
Is this a problem if `url patterns` is null from step 2? Should we default to empty list instead of null?

Reply:
Was just implementing this, and yes, empty list would be more convenient.